streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorderTest.java

```diff
@@ -300,65 +287,47 @@ public void shouldAddItselfToRecordingTriggerWhenFirstValueProvidersAreAddedAfte
         recorder.removeValueProviders(SEGMENT_STORE_NAME_1);
         reset(recordingTrigger);
         recordingTrigger.addMetricsRecorder(recorder);
-        replay(recordingTrigger);
 
         recorder.addValueProviders(SEGMENT_STORE_NAME_2, dbToAdd2, cacheToAdd2, statisticsToAdd2);
-
-        verify(recordingTrigger);
     }
 
     @Test
     public void shouldNotAddItselfToRecordingTriggerWhenNotEmpty2() {
         recorder.addValueProviders(SEGMENT_STORE_NAME_1, dbToAdd1, cacheToAdd1, statisticsToAdd1);
         reset(recordingTrigger);
-        replay(recordingTrigger);
 
         recorder.addValueProviders(SEGMENT_STORE_NAME_2, dbToAdd2, cacheToAdd2, statisticsToAdd2);
-
-        verify(recordingTrigger);
     }
 
     @Test
     public void shouldCloseStatisticsWhenValueProvidersAreRemoved() {
         recorder.addValueProviders(SEGMENT_STORE_NAME_1, dbToAdd1, cacheToAdd1, statisticsToAdd1);
         reset(statisticsToAdd1);
         statisticsToAdd1.close();
-        replay(statisticsToAdd1);
 
         recorder.removeValueProviders(SEGMENT_STORE_NAME_1);
-
-        verify(statisticsToAdd1);
     }
 
     @Test
     public void shouldNotCloseStatisticsWhenValueProvidersWithoutStatisticsAreRemoved() {
         recorder.addValueProviders(SEGMENT_STORE_NAME_1, dbToAdd1, cacheToAdd1, null);
         reset(statisticsToAdd1);
-        replay(statisticsToAdd1);
 
         recorder.removeValueProviders(SEGMENT_STORE_NAME_1);
```

Comment:
Please using Mockito.verify to make sure recorder.removeValueProviders(SEGMENT_STORE_NAME_2); is not executed when executing recorder.removeValueProviders(SEGMENT_STORE_NAME_2);